src/components/popup/PopupWithForm.js

```diff
@@ -0,0 +1,34 @@
+import { Link } from 'react-router-dom';
+import * as React from 'react';
+
+export default function PopupWithForm(props) {
+	const { linkText, name, title, onSubmit, children, isValid, handleSwitchPopup, buttonText, isOpen, onClose } = props;
+	
+	// ! Switching between popups
+	const handleLinkClick = () => {
+		onClose();
+		handleSwitchPopup();
+	}
+
+	// ! Reseting the form
+	React.useEffect(() => {
+		const form = document.querySelector(`.form-${name}`);
```

Comment:
```js
		const form = document.querySelector(`.form-${name}`);
		form.reset();

```


please, delete it


you have  `controlled` forms, you you can't clear them in React with `reset`


You should use the state functions from `useState` that control the values

For example:

```js
    React.useEffect(() => {
        setName('');
        setLink('');
    }, [isOpen]);

```